src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    
+    Animal.alive.push(this);
+  }
+
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(animal) {
+    if (animal instanceof Carnivore || animal.hidden) {
```

Comment:
The check `animal.hidden` will cause an error if `animal` is an instance of Carnivore, as the `hidden` property does not exist on Carnivore instances. It's better to check if `animal` is an instance of Herbivore first and then check if it is hidden.